bucket.go

```diff
@@ -73,7 +73,12 @@ func (b *Bucket) Writable() bool {
 // Do not use a cursor after the transaction is closed.
 func (b *Bucket) Cursor() *Cursor {
 	// Update transaction statistics.
-	b.tx.stats.IncCursorCount(1)
+	// Note that `Cursor()` may be called by multiple readonly
+	// transactions concurrently, so we need to hold the lock
+	// `statlock` when updating the stats.
+	b.tx.db.statlock.Lock()
```

Comment:
I understand that's very nitpicky, but do you know what the perf difference here would be? The cursor is pretty widely used and just taking a lock to increment an int seems a bit much.